src/controllers/orderController.js

```diff
@@ -0,0 +1,48 @@
+const { Order, OrderItem } = require('../models/Order');
+const { StatusCodes } = require('http-status-codes');
+const stripe = require('stripe')(process.env.STRIPE_SECRET_KEY);
+const { BadRequestError } = require('../errors');
+
+const createOrder = async (req, res) => {
+  const { orderItems, subtotal, tax } = req.body;
+
+  try {
+    // Create an array of OrderItem objects
+    const orderItemObjects = orderItems.map(item => new OrderItem(item));
+    // Calculate the total price including tax
+    const total = subtotal + (subtotal * tax);
+
+    // Perform necessary checks and validations
+    if (orderItems.length === 0) {
+      throw new BadRequestError('Unable to create order because the order items are missing.'); 
+    }
+
+    // Create an Order document in the database
+    const order = await Order.create({
+      user: req.user.userId,
+      orderItems: orderItemObjects,
+      subtotal,
+      tax,
+      total,
+      orderStatus: 'pending', // Set initial order status as pending
+    });
+
+    // Create a payment intent with Stripe
+    const paymentIntent = await stripe.paymentIntents.create({
+      amount: total * 100, // Stripe requires amount in cents
+      currency: 'usd',
+      description: `Order #${order._id}`,
+    });
+
+    // Send the payment intent client secret and order information to the client
```

Comment:
We need to also save the `paymentIntent` to our Order entry in the db to keep it on record (and later we can also get live updates about it from Stripe via webhooks).

`order.paymentId = ..., order.save`

Reply:
// Save the paymentIntent to the Order model
  order.paymentIntentId = paymentIntent.id;
  await order.save();